dataprofiler/profilers/profile_builder.py

```diff
@@ -1228,17 +1237,76 @@ def _min_col_samples_used(self):
         i.e. every column in the Profile was read up to this row (possibly
         further in some cols)
         """
-        return min([self._profile[col].sample_size
-                    for col in self._profile], default=0)
+        return min([col.sample_size for col in self._profile], default=0)
 
     @property
     def _min_sampled_from_batch(self):
         """
         Calculates and returns the number of rows that were completely sampled
         in the most previous batch
         """
-        return min([self._profile[col]._last_batch_size
-                    for col in self._profile], default=0)
+        return min([col._last_batch_size for col in self._profile], default=0)
+
+    @staticmethod
+    def _get_and_validate_schema_mapping(schema1, schema2, strict=False):
+        """
+        Validate compatibility between schema1 and schema2 and return a dict
+        mapping indices in schema1 to their corresponding indices in schema2.
+        In __add__: want to map self _profile idx -> other _profile idx
+        In _update_profile_from_chunk: want to map data idx -> _profile idx
+
+        :param schema1: a column name to index mapping
+        :type schema1: Dict[str, list[int]]
+        :param schema2: a column name to index mapping
+        :type schema2: Dict[str, list[int]]
+        :param strict: whether or not to strictly match (__add__ case)
+        :type strict: bool
+        :return: a mapping of indices in schema1 to indices in schema2
+        :rtype: Dict[int, int]
+        """
+
+        len_schema1 = len(schema1)
+        len_schema2 = len(schema2)
+
+        # If both non-empty, must be same length
+        if 0 < len_schema1 != len_schema2 > 0:
```

Comment:
woah. Cool

Reply:
Yeah I originally had it as 3 separate `and` conditions and PyCharm told me I could combine them. Btw, this PR has already basically been completely approved except for the chunks that I commented "This is new", so you can focus your review on just those parts. Everything else was already approved into the `duplicate_column_fix` branch. Just letting you know cuz this PR is like 600 lines but only like 60 of them really need review.